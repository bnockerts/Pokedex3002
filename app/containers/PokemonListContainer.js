import React, { Component } from 'react';
import { connect } from 'react-redux';
import Spinner from '../components/Spinner';
import PokemonList from '../components/PokemonList';
import { navigatePush, fetchPokemonIfNeeded, toggleCaught } from '../actions';
import {
  ListView
} from 'react-native';

class PokemonListContainer extends Component {
	  componentDidMount() {
			this.props.loadData();
		}

		render() {
			if (this.props.isLoading) {
				return (
					<Spinner isLoading={this.props.isLoading} />
				);
			}

			return (
				<PokemonList {...this.props} />
			)
  }
}

const mapStateToProps = (state) => {
	const { pokemonList } = state;
  const { isLoading, pokemon } = pokemonList;

  return {
    isLoading,
    pokemon
  };
}

const mapDispatchToProps = (dispatch) => {
	return {
		loadData: () => {
			dispatch(fetchPokemonIfNeeded());
		},

		onValueChange: (id) => {
			dispatch(toggleCaught(id));
		},

		onRowPress: (pokemon) => {
			dispatch(navigatePush({
				key: 'PokemonDetail',
				title: pokemon.name,
				pokemon
			}));
		}
	};
}

export default connect(mapStateToProps, mapDispatchToProps)(PokemonListContainer);
