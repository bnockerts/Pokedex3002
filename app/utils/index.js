const pokemonListData = { "count": 811, "previous": null, "results": [{ "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/1\/", "name": "bulbasaur" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/2\/", "name": "ivysaur" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/3\/", "name": "venusaur" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/4\/", "name": "charmander" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/5\/", "name": "charmeleon" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/6\/", "name": "charizard" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/7\/", "name": "squirtle" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/8\/", "name": "wartortle" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/9\/", "name": "blastoise" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/10\/", "name": "caterpie" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/11\/", "name": "metapod" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/12\/", "name": "butterfree" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/13\/", "name": "weedle" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/14\/", "name": "kakuna" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/15\/", "name": "beedrill" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/16\/", "name": "pidgey" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/17\/", "name": "pidgeotto" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/18\/", "name": "pidgeot" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/19\/", "name": "rattata" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/20\/", "name": "raticate" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/21\/", "name": "spearow" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/22\/", "name": "fearow" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/23\/", "name": "ekans" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/24\/", "name": "arbok" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/25\/", "name": "pikachu" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/26\/", "name": "raichu" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/27\/", "name": "sandshrew" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/28\/", "name": "sandslash" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/29\/", "name": "nidoran-f" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/30\/", "name": "nidorina" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/31\/", "name": "nidoqueen" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/32\/", "name": "nidoran-m" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/33\/", "name": "nidorino" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/34\/", "name": "nidoking" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/35\/", "name": "clefairy" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/36\/", "name": "clefable" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/37\/", "name": "vulpix" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/38\/", "name": "ninetales" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/39\/", "name": "jigglypuff" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/40\/", "name": "wigglytuff" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/41\/", "name": "zubat" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/42\/", "name": "golbat" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/43\/", "name": "oddish" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/44\/", "name": "gloom" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/45\/", "name": "vileplume" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/46\/", "name": "paras" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/47\/", "name": "parasect" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/48\/", "name": "venonat" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/49\/", "name": "venomoth" }, { "url": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/50\/", "name": "diglett" }], "next": "https:\/\/pokeapi.co\/api\/v2\/pokemon\/?limit=50&offset=50" };

const pokemonDetailData = {
  forms: [{
      url: 'https://pokeapi.co/api/v2/pokemon-form/6/',
      name: 'charizard'
  }],
  abilities: [{
      slot: 3,
      is_hidden: true,
      ability: {
          url: 'https://pokeapi.co/api/v2/ability/94/',
          name: 'solar-power'
      }
  }, {
      slot: 1,
      is_hidden: false,
      ability: { url: 'https://pokeapi.co/api/v2/ability/66/', name: 'blaze' }
  }],
  stats: [{
      stat: { url: 'https://pokeapi.co/api/v2/stat/6/', name: 'speed' },
      effort: 0,
      base_stat: 100
  }, {
      stat: {
          url: 'https://pokeapi.co/api/v2/stat/5/',
          name: 'special-defense'
      },
      effort: 0,
      base_stat: 85
  }, {
      stat: {
          url: 'https://pokeapi.co/api/v2/stat/4/',
          name: 'special-attack'
      },
      effort: 3,
      base_stat: 109
  }, {
      stat: { url: 'https://pokeapi.co/api/v2/stat/3/', name: 'defense' },
      effort: 0,
      base_stat: 78
  }, {
      stat: { url: 'https://pokeapi.co/api/v2/stat/2/', name: 'attack' },
      effort: 0,
      base_stat: 84
  }, {
      stat: { url: 'https://pokeapi.co/api/v2/stat/1/', name: 'hp' },
      effort: 0,
      base_stat: 78
  }],
  name: 'charizard',
  weight: 905,
  sprites: {
      back_female: null,
      back_shiny_female: null,
      back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png',
      front_female: null,
      front_shiny_female: null,
      back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png',
      front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png',
      front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png'
  },
  held_items: [],
  location_area_encounters: '/api/v2/pokemon/6/encounters',
  height: 17,
  is_default: true,
  species: {
      url: 'https://pokeapi.co/api/v2/pokemon-species/6/',
      name: 'charizard'
  },
  id: 6,
  order: 7,
  game_indices: [{
      version: {
          url: 'https://pokeapi.co/api/v2/version/22/',
          name: 'white-2'
      },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/21/',
          name: 'black-2'
      },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/18/', name: 'white' },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/17/', name: 'black' },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/16/',
          name: 'soulsilver'
      },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/15/',
          name: 'heartgold'
      },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/14/',
          name: 'platinum'
      },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/13/', name: 'pearl' },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/12/',
          name: 'diamond'
      },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/11/',
          name: 'leafgreen'
      },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/10/',
          name: 'firered'
      },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/9/', name: 'emerald' },
      game_index: 6
  }, {
      version: {
          url: 'https://pokeapi.co/api/v2/version/8/',
          name: 'sapphire'
      },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/7/', name: 'ruby' },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/6/', name: 'crystal' },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/5/', name: 'silver' },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/4/', name: 'gold' },
      game_index: 6
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/3/', name: 'yellow' },
      game_index: 180
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/2/', name: 'blue' },
      game_index: 180
  }, {
      version: { url: 'https://pokeapi.co/api/v2/version/1/', name: 'red' },
      game_index: 180
  }],
  base_experience: 240,
  types: [{
      slot: 2,
      type: { url: 'https://pokeapi.co/api/v2/type/3/', name: 'flying' }
  }, {
      slot: 1,
      type: { url: 'https://pokeapi.co/api/v2/type/10/', name: 'fire' }
  }]
};

export {
  pokemonListData,
  pokemonDetailData
}
