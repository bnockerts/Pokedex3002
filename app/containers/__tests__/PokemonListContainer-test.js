import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';
import { PokemonListContainer } from '../PokemonListContainer';
import { formattedPokemonListData } from '../../utils';

const shallowRenderer = ReactTestUtils.createRenderer();

it('renders a PokemonListContainer', () => {
  const testProps = {
    onRowPress: jest.fn,
    onValueChange: jest.fn,
    pokemon: formattedPokemonListData
  };

  const component = shallowRenderer.render(
    <PokemonListContainer {...testProps} />
  );

  expect(component).toMatchSnapshot();
});
