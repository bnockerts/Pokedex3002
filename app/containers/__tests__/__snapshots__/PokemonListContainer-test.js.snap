exports[`test renders a PokemonListContainer 1`] = `
<PokemonList
  onRowPress={[Function]}
  onValueChange={[Function]}
  pokemon={
    Array [
      Object {
        "id": "1",
        "name": "Bulbasaur",
        "url": "https://pokeapi.co/api/v2/pokemon/1/",
      },
      Object {
        "id": "2",
        "name": "Ivysaur",
        "url": "https://pokeapi.co/api/v2/pokemon/2/",
      },
      Object {
        "id": "3",
        "name": "Venusaur",
        "url": "https://pokeapi.co/api/v2/pokemon/3/",
      },
      Object {
        "id": "4",
        "name": "Charmander",
        "url": "https://pokeapi.co/api/v2/pokemon/4/",
      },
      Object {
        "id": "5",
        "name": "Charmeleon",
        "url": "https://pokeapi.co/api/v2/pokemon/5/",
      },
      Object {
        "id": "6",
        "name": "Charizard",
        "url": "https://pokeapi.co/api/v2/pokemon/6/",
      },
      Object {
        "id": "7",
        "name": "Squirtle",
        "url": "https://pokeapi.co/api/v2/pokemon/7/",
      },
      Object {
        "id": "8",
        "name": "Wartortle",
        "url": "https://pokeapi.co/api/v2/pokemon/8/",
      },
      Object {
        "id": "9",
        "name": "Blastoise",
        "url": "https://pokeapi.co/api/v2/pokemon/9/",
      },
      Object {
        "id": "10",
        "name": "Caterpie",
        "url": "https://pokeapi.co/api/v2/pokemon/10/",
      },
      Object {
        "id": "11",
        "name": "Metapod",
        "url": "https://pokeapi.co/api/v2/pokemon/11/",
      },
      Object {
        "id": "12",
        "name": "Butterfree",
        "url": "https://pokeapi.co/api/v2/pokemon/12/",
      },
      Object {
        "id": "13",
        "name": "Weedle",
        "url": "https://pokeapi.co/api/v2/pokemon/13/",
      },
      Object {
        "id": "14",
        "name": "Kakuna",
        "url": "https://pokeapi.co/api/v2/pokemon/14/",
      },
      Object {
        "id": "15",
        "name": "Beedrill",
        "url": "https://pokeapi.co/api/v2/pokemon/15/",
      },
      Object {
        "id": "16",
        "name": "Pidgey",
        "url": "https://pokeapi.co/api/v2/pokemon/16/",
      },
      Object {
        "id": "17",
        "name": "Pidgeotto",
        "url": "https://pokeapi.co/api/v2/pokemon/17/",
      },
      Object {
        "id": "18",
        "name": "Pidgeot",
        "url": "https://pokeapi.co/api/v2/pokemon/18/",
      },
      Object {
        "id": "19",
        "name": "Rattata",
        "url": "https://pokeapi.co/api/v2/pokemon/19/",
      },
      Object {
        "id": "20",
        "name": "Raticate",
        "url": "https://pokeapi.co/api/v2/pokemon/20/",
      },
      Object {
        "id": "21",
        "name": "Spearow",
        "url": "https://pokeapi.co/api/v2/pokemon/21/",
      },
      Object {
        "id": "22",
        "name": "Fearow",
        "url": "https://pokeapi.co/api/v2/pokemon/22/",
      },
      Object {
        "id": "23",
        "name": "Ekans",
        "url": "https://pokeapi.co/api/v2/pokemon/23/",
      },
      Object {
        "id": "24",
        "name": "Arbok",
        "url": "https://pokeapi.co/api/v2/pokemon/24/",
      },
      Object {
        "id": "25",
        "name": "Pikachu",
        "url": "https://pokeapi.co/api/v2/pokemon/25/",
      },
      Object {
        "id": "26",
        "name": "Raichu",
        "url": "https://pokeapi.co/api/v2/pokemon/26/",
      },
      Object {
        "id": "27",
        "name": "Sandshrew",
        "url": "https://pokeapi.co/api/v2/pokemon/27/",
      },
      Object {
        "id": "28",
        "name": "Sandslash",
        "url": "https://pokeapi.co/api/v2/pokemon/28/",
      },
      Object {
        "id": "29",
        "name": "Nidoran-f",
        "url": "https://pokeapi.co/api/v2/pokemon/29/",
      },
      Object {
        "id": "30",
        "name": "Nidorina",
        "url": "https://pokeapi.co/api/v2/pokemon/30/",
      },
      Object {
        "id": "31",
        "name": "Nidoqueen",
        "url": "https://pokeapi.co/api/v2/pokemon/31/",
      },
      Object {
        "id": "32",
        "name": "Nidoran-m",
        "url": "https://pokeapi.co/api/v2/pokemon/32/",
      },
      Object {
        "id": "33",
        "name": "Nidorino",
        "url": "https://pokeapi.co/api/v2/pokemon/33/",
      },
      Object {
        "id": "34",
        "name": "Nidoking",
        "url": "https://pokeapi.co/api/v2/pokemon/34/",
      },
      Object {
        "id": "35",
        "name": "Clefairy",
        "url": "https://pokeapi.co/api/v2/pokemon/35/",
      },
      Object {
        "id": "36",
        "name": "Clefable",
        "url": "https://pokeapi.co/api/v2/pokemon/36/",
      },
      Object {
        "id": "37",
        "name": "Vulpix",
        "url": "https://pokeapi.co/api/v2/pokemon/37/",
      },
      Object {
        "id": "38",
        "name": "Ninetales",
        "url": "https://pokeapi.co/api/v2/pokemon/38/",
      },
      Object {
        "id": "39",
        "name": "Jigglypuff",
        "url": "https://pokeapi.co/api/v2/pokemon/39/",
      },
      Object {
        "id": "40",
        "name": "Wigglytuff",
        "url": "https://pokeapi.co/api/v2/pokemon/40/",
      },
      Object {
        "id": "41",
        "name": "Zubat",
        "url": "https://pokeapi.co/api/v2/pokemon/41/",
      },
      Object {
        "id": "42",
        "name": "Golbat",
        "url": "https://pokeapi.co/api/v2/pokemon/42/",
      },
      Object {
        "id": "43",
        "name": "Oddish",
        "url": "https://pokeapi.co/api/v2/pokemon/43/",
      },
      Object {
        "id": "44",
        "name": "Gloom",
        "url": "https://pokeapi.co/api/v2/pokemon/44/",
      },
      Object {
        "id": "45",
        "name": "Vileplume",
        "url": "https://pokeapi.co/api/v2/pokemon/45/",
      },
      Object {
        "id": "46",
        "name": "Paras",
        "url": "https://pokeapi.co/api/v2/pokemon/46/",
      },
      Object {
        "id": "47",
        "name": "Parasect",
        "url": "https://pokeapi.co/api/v2/pokemon/47/",
      },
      Object {
        "id": "48",
        "name": "Venonat",
        "url": "https://pokeapi.co/api/v2/pokemon/48/",
      },
      Object {
        "id": "49",
        "name": "Venomoth",
        "url": "https://pokeapi.co/api/v2/pokemon/49/",
      },
      Object {
        "id": "50",
        "name": "Diglett",
        "url": "https://pokeapi.co/api/v2/pokemon/50/",
      },
    ]
  } />
`;
